src/Smuggler/Name.hs

```diff
@@ -0,0 +1,11 @@
+module Smuggler.Name
+       ( moduleBodyNames
+       ) where
+
+import Data.Generics.Schemes (listify)
+import GHC (GenLocated (..), ParsedSource)
+import HsSyn (HsModule (..))
+import OccName (OccName)
+
+moduleBodyNames :: ParsedSource -> [OccName]
+moduleBodyNames (L _ ast) = ordNub $ listify (\(_ :: OccName) -> True) (hsmodDecls ast)
```

Comment:
That's really cool! 
I'm not very familiar with this approach, so you're just taking all `OccName`s down the road and put it in the list, is my understanding correct?

Reply:
@vrom911 Yeas, your understanding is correct. Functions from `syb` (Scrap Your Boilerplate) package rely on `Data` instances for data types. And there're different functions. Basically, what `listify` does is traverses all structure and collects all values of type `OccName` into single list. This is very convenient, because:

1. I don't know what are all places for `OccName`.
2. I don't need to write manual pattern matching for every case.

This approach is slower than actual manual pattern-matching. But I think that in our case nobody will notice difference. And it's better to rely on existing machinery.